wasmcloud-test-util/src/lib.rs

```diff
@@ -4,13 +4,65 @@ pub mod provider_test;
 #[cfg(not(target_arch = "wasm32"))]
 pub mod cli;
 
-// re-export testing interface
-pub use wasmcloud_interface_testing as testing;
-
 // re-export regex and nkeys
 pub use nkeys;
 pub use regex;
 
+#[allow(dead_code)]
+pub mod testing {
+
+    smithy_bindgen::smithy_bindgen!("testing/testing.smithy", "org.wasmcloud.interface.testing");
+
+    // after sdk is split we won't have to duplicate this code
+    impl Default for TestOptions {
+        fn default() -> TestOptions {
+            TestOptions {
+                patterns: vec![".*".to_string()],
+                options: std::collections::HashMap::default(),
+            }
+        }
+    }
+
+    pub type NamedResult<'name, T> = (&'name str, RpcResult<T>);
+
+    // convert empty RpcResult into a testResult
+    impl<'name, T: Serialize> From<NamedResult<'name, T>> for TestResult {
+        fn from(name_res: NamedResult<'name, T>) -> TestResult {
+            match name_res.1 {
```

Comment:
How about using a destructor instead?

Reply:
I'm not sure what you mean .. if it's empty we still want to return a result

Reply:
For example, https://github.com/wasmCloud/wasmcloud-test/pull/24/files#r1118852182

Reply:
I was referring to https://github.com/rust-lang/rfcs/pull/2909

Reply:
ah, "destructuring assignment".
Good suggestion, done.